libraries/apollo-api/src/commonMain/kotlin/com/apollographql/apollo/api/http/DefaultHttpRequestComposer.kt

```diff
@@ -50,19 +50,20 @@ class DefaultHttpRequestComposer(
     }
 
     val sendApqExtensions = apolloRequest.sendApqExtensions ?: false
+    val sendEnhancedClientAwarenessExtensions = true
```

Comment:
This obviously needs to get it from somewhere; I _think_ we'd discussed setting the value at client creation?

Reply:
Right. This is both one of the most boring and one of the most complicated part of the codebase. 

I would recommend adding a property to `ApolloClient.Builder` like `failFastIfOnline` [here](https://github.com/apollographql/apollo-kotlin/blob/4b6dbc6750672faab897656a3651a655b99038fe/libraries/apollo-runtime/src/commonMain/kotlin/com/apollographql/apollo/ApolloClient.kt#L391). The other option is to be able to set it individually per request in `MutableExecutionOptions` but I don't think it's worth the added complexity.